integration/utmp_integration_test.go

```diff
@@ -115,6 +115,18 @@ func TestRootUTMPEntryExists(t *testing.T) {
 	t.Errorf("did not detect utmp entry within 5 minutes")
 }
 
+// TestUsernameLimit tests that the maximum length of usernames is a hard error.
+func TestUsernameLimit(t *testing.T) {
+	dir := t.TempDir()
+
+	// A 33 character long username.
+	username := "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+	host := [4]int32{0, 0, 0, 0}
+	tty := os.NewFile(uintptr(5), "/proc/self/fd/0")
+	err := uacc.Open(dir+"/utmp", dir+"/wtmp", username, "localhost", host, tty)
```

Comment:
any way to make these paths portable (or skip the test on systems where it's just not going to work)?

Reply:
This test never runs on another platform anyway. The utmp stuff needs a special make flag and CI only runs it on Linux anyway and has a specific section for it. Calling the uacc functions on any other platform results in a noop as the functionality only compiles on linux and all other platforms gets a stub file.

Reply:
ahh makes sense